src/goto-instrument/contracts/assigns.cpp

```diff
@@ -145,11 +170,15 @@ exprt assigns_clauset::generate_containment_check(const exprt &lhs) const
 exprt assigns_clauset::generate_subset_check(
   const assigns_clauset &subassigns) const
 {
-  if(subassigns.targets.empty())
+  if(subassigns.global_write_set.empty())
     return true_exprt();
 
+  std::unordered_set<targett, targett::hasht> caller_write_set(
+    global_write_set);
+  caller_write_set.insert(local_write_set.begin(), local_write_set.end());
+
```

Comment:
I think it might be cleaner (and more efficient) to just have another loop. The loop only has a single statement in it anyway. What do you think?

Reply:
Implemented as suggested.